source/hash_table.c

```diff
@@ -542,13 +550,14 @@ static size_t s_remove_entry(struct hash_table_state *state, struct hash_table_e
         }
 
         /* Okay, shift this one back */
-        memcpy(&state->slots[index], &state->slots[next_index], sizeof(*state->slots));
+        state->slots[index] = state->slots[next_index];
```

Comment:
It's not obvious to me why you made this change. Does this fix a bug, or is it a semantics-preserving change that makes the proof possible or faster? Please add an explanation to the commit message

Reply:
This is semantics preserving.  In my mind, it has two advantages: 
1. A simple assignment is more obviously correct than the memcpy (where an ```&``` or a ```*``` could easily be added or missed)
2. In some cases, I want the ability to override ```memcpy``` as part of CBMC proofs. The fewer times ```memcpy``` appears, the less I/CBMC have to reason about what it is doing.